### PR TITLE
Handle invalid UUID on queries

### PR DESCRIPTION
**Pull request by Dhwaniartefact**
Connected to https://github.com/archivematica/Issues/issues/1621

### CODE REVIEW

File: src/MCPClient/lib/clientScripts/create_sip_from_transfer_objects.py [hunk 1]
```diff
@@ -117,12 +118,12 @@ def call(jobs):
...
                         variable="activeAgent",
                     )
-                except UnitVariable.DoesNotExist:
+                except (UnitVariable.DoesNotExist, ValidationError):
                     unit_variable = None
                 if unit_variable:
                     try:
                         agent = Agent.objects.get(id=unit_variable.variablevalue)
-                    except Agent.DoesNotExist:
+                    except (Agent.DoesNotExist, ValidationError):
```
Comment (replaceafill) [inline src/MCPClient/lib/clientScripts/create_sip_from_transfer_objects.py]: Is this one necessary? The `Agent.id` field is not a `UUIDField`.

File: src/MCPClient/lib/clientScripts/create_transfer_mets.py [hunk 2]
```diff
@@ -84,8 +85,8 @@ def write_mets(mets_path, transfer_dir_path, base_path_placeholder, transfer_uui
         mets.agents.append(agent)
 
     try:
-        transfer = Transfer.objects.get(uuid=transfer_uuid)
-    except Transfer.DoesNotExist:
+        transfer = Transfer.objects.get(uuid=str(transfer_uuid))
```
Comment (replaceafill) [inline src/MCPClient/lib/clientScripts/create_transfer_mets.py]: Is the `str(...)` needed?

File: src/MCPClient/lib/clientScripts/dip_generation_helper.py [hunk 3]
```diff
@@ -6,10 +6,11 @@
...
 from custom_handlers import get_script_logger
+from django.core.exceptions import ValidationError
 from django.db.models import Q
 from main import models
 
-# dashboard
+# dashboardk
 # archivematicaCommon
 # Third party dependencies, alphabetical by import source
 # initialize Django (required for Django 1.7)
```
Comment (replaceafill) [inline src/MCPClient/lib/clientScripts/dip_generation_helper.py]: Could you remove all these comments?

File: src/MCPClient/lib/clientScripts/parse_mets_to_db.py [hunk 4]
```diff
@@ -44,7 +45,7 @@ def parse_format_version(job, element):
             key = ns.xml_findtext_premis(element, ".//premis:formatRegistryKey")
             job.pyprint("FPR key", key)
             format_version = fpr_models.IDRule.active.get(command_output=key).format
-    except fpr_models.FormatVersion.DoesNotExist:
+    except (fpr_models.FormatVersion.DoesNotExist, ValidationError):
```
Comment (replaceafill) [inline src/MCPClient/lib/clientScripts/parse_mets_to_db.py]: Is this necessary?

File: src/dashboard/src/components/access/views.py [hunk 5]
```diff
@@ -72,7 +73,7 @@ def wrapper(request, record_id=""):
                 identifier=_normalize_record_id(record_id),
             )
             return func(request, mapping)
-        except SIPArrangeAccessMapping.DoesNotExist:
+        except (SIPArrangeAccessMapping.DoesNotExist, ValidationError):
```
Comment (replaceafill) [inline src/dashboard/src/components/access/views.py]: Is this necessary?

File: src/dashboard/src/components/access/views.py [hunk 6]
```diff
@@ -165,7 +166,7 @@ def record(client, request, record_id=""):
                 system=SIPArrangeAccessMapping.ARCHIVESSPACE,
                 identifier=_normalize_record_id(record_id),
             )
-        except SIPArrangeAccessMapping.DoesNotExist:
+        except (SIPArrangeAccessMapping.DoesNotExist, ValidationError):
```
Comment (replaceafill) [inline src/dashboard/src/components/access/views.py]: Is this necessary?

File: src/dashboard/src/components/access/views.py [hunk 7]
```diff
@@ -285,7 +286,7 @@ def digital_object_components(client, request, record_id=""):
 
         try:
             component = ArchivesSpaceDigitalObject.objects.get(id=component_id)
-        except ArchivesSpaceDigitalObject.DoesNotExist:
+        except (ArchivesSpaceDigitalObject.DoesNotExist, ValidationError):
```
Comment (replaceafill) [inline src/dashboard/src/components/access/views.py]: Is this necessary?

File: src/dashboard/src/components/access/views.py [hunk 8]
```diff
@@ -312,6 +313,7 @@ def _get_sip_arrange_children(record):
     except (
         SIPArrangeAccessMapping.MultipleObjectsReturned,
         SIPArrangeAccessMapping.DoesNotExist,
+        ValidationError,
```
Comment (replaceafill) [inline src/dashboard/src/components/access/views.py]: Is this necessary?

File: src/dashboard/src/components/access/views.py [hunk 9]
```diff
@@ -423,7 +425,7 @@ def access_arrange_start_sip(client, request, mapping):
         arrange = SIPArrange.objects.get(
             arrange_path=os.path.join(mapping.arrange_path, "")
         )
-    except SIPArrange.DoesNotExist:
+    except (SIPArrange.DoesNotExist, ValidationError):
```
Comment (replaceafill) [inline src/dashboard/src/components/access/views.py]: Is this necessary?

File: src/dashboard/src/components/transfer/views.py [hunk 10]
```diff
@@ -99,7 +100,7 @@ def component(request, uuid):
                 field=field, set=metadata_set
             )
             values[(field.fieldname)] = field_value.fieldvalue
-        except models.TransferMetadataFieldValue.DoesNotExist:
+        except (models.TransferMetadataFieldValue.DoesNotExist, ValidationError):
```
Comment (replaceafill) [inline src/dashboard/src/components/transfer/views.py]: Is this necessary?

File: src/dashboard/src/fpr/utils.py [hunk 11]
```diff
@@ -102,7 +102,7 @@ def get_revision_descendants(model, uuid, decendants):
     revision = model.objects.get(uuid=uuid)
     try:
         descendant = model.objects.get(replaces=revision)
-    except model.DoesNotExist:
+    except (model.DoesNotExist, ValidationError):
```
Comment (replaceafill) [inline src/dashboard/src/fpr/utils.py]: Is this necessary?

**Review by replaceafill — CHANGES_REQUESTED** [review]
Looks good @Dhwaniartefact, just left some questions.

File: src/dashboard/src/fpr/utils.py (hunk 11 above)
Comment (Dhwaniartefact) [reply src/dashboard/src/fpr/utils.py]: Thanks you for all your suggestions and will make those changes accordingly.

**Review by replaceafill — APPROVED** [review]

**Pull request merged**